provider/pkg/provider/provider.go

```diff
@@ -407,7 +416,44 @@ func (p *googleCloudProvider) Diff(_ context.Context, req *rpc.DiffRequest) (*rp
 		return &rpc.DiffResponse{Changes: rpc.DiffResponse_DIFF_NONE}, nil
 	}
 
-	return &rpc.DiffResponse{Changes: rpc.DiffResponse_DIFF_UNKNOWN, DeleteBeforeReplace: true}, nil
+	// Calculate the detailed diff object containing information about replacements.
+	detailedDiff := calculateDetailedDiff(&res, p.resourceMap.Types, diff)
+
+	// Based on the detailed diff above, calculate the list of changes and replacements.
+	var changes, replaces []string
+	for k, v := range detailedDiff {
+		parts := strings.Split(k, ".")
+		changes = append(changes, parts[0])
+		v.InputDiff = true
+
+		switch v.Kind {
+		case rpc.PropertyDiff_ADD_REPLACE:
+			replaces = append(replaces, k)
+		case rpc.PropertyDiff_DELETE_REPLACE, rpc.PropertyDiff_UPDATE_REPLACE:
+			replaces = append(replaces, k)
+		}
+	}
+
+	deleteBeforeReplace := len(replaces) > 0
+	if v, ok := oldInputs[createBeforeDeleteFlag]; ok && v.IsBool() {
+		deleteBeforeReplace = !v.BoolValue()
```

Comment:
Would it make sense to name this field `__deleteBeforeReplace` to avoid having to invert the logic here? I'm not sure if `createBeforeDelete` is the exact inverse of `deleteBeforeReplace`?

Reply:
Thanks for pointing out - this is copied from azure-native. I changed the semantics to make this a little easier/clearer. Let me know what you think.

Reply:
Much clearer, thanks!